e2e/launcher/pocketclient.go

```diff
@@ -0,0 +1,55 @@
+package launcher
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+type CommandResult struct {
+	Stdout string
+	Stderr string
+	Err    error
+}
+
+type writerByteArray []byte
+
+func (so *writerByteArray) Write(p []byte) (n int, err error) {
+	*so = append(*so, p...)
+	return len(p), nil
+}
+
+type PocketClient interface {
+	RunCommand(...string) (*CommandResult, error)
+}
+
+func NewPocketClient(executableLocation string, verbose bool) PocketClient {
+	return &pocketClient{
+		executableLocation: executableLocation,
+		verbose:            verbose,
+	}
+}
+
+type pocketClient struct {
+	executableLocation string
+	verbose            bool
+}
+
+func (pc *pocketClient) RunCommand(commandAndArgs ...string) (*CommandResult, error) {
+	if pc.verbose {
+		fmt.Printf("Running Command: %v\n", commandAndArgs)
+	}
+	cmd := exec.Command(pc.executableLocation, commandAndArgs...)
+
+	so := &writerByteArray{}
+	se := &writerByteArray{}
+
+	cmd.Stdout = so
+	cmd.Stderr = se
+	err := cmd.Run()
+
+	return &CommandResult{
+		Stdout: string(*so),
```

Comment:
Since we're using the byte array to keep appending and convert it to a string in the end, thoughts on using the strings builder in golang? It implements `Write` by default: https://pkg.go.dev/strings#Builder.Write which is the interface `Stdout`  above needs.